Zend/zend_alloc.c

```diff
@@ -3092,199 +3082,111 @@ ZEND_API void zend_mm_get_custom_handlers(zend_mm_heap *heap,
 }
 
 /*
- * Use this function to register observers to the ZendMM. Make sure to call this
- * function only during PHP module startup phase and not later. Observers will
- * not be called directly after registering them, but only after
- * `zend_mm_observers_startup()` was run, which will take place after the PHP
- * module startup phase.
+ * Use this function to register observers to the ZendMM.
+ * This function operates on the thread local heap and is meant to be called in
+ * RINIT. Calling it in MINIT works only in NTS builds, but not in ZTS.
```

Comment:
Can we also call it during a request after RINIT? It looks like we could, but I may be missing something

Reply:
Yes we can, I'll have to update the comment. I meant to write that one can not reliably register before RINIT, but you can expose this functionality to the user space and register/unregister at any point between RINIT and RSHUTDOWN